L89-core-remove-grpc-insecure-channel-creation-api

```diff
@@ -0,0 +1,128 @@
+Remove grpc_insecure_channel_create and grpc_server_add_insecure_http2_port from Core Surface API
+----
+* Author(s): yihuazhang
+* Approver: markdroth
+* Status: {In Review}
+* Implemented in: https://github.com/grpc/grpc/pull/25586
+* Last updated: 2021-11-01
+* Discussion at: https://groups.google.com/g/grpc-io/c/9B4-AHCx_ZM 
+
+## Abstract
+
+* Remove `grpc_insecure_channel_create` and `grpc_server_add_insecure_http2_port` from Core Surface API.
+* Rename `grpc_server_add_secure_http2_port` to `grpc_server_add_http2_port`.
+* Rename `grpc_secure_channel_create` to `grpc_channel_create` and re-order its parameters.
+* Move `grpc_channel_credentials` and `grpc_server_credentials`related API's from `grpc_security.h` to `grpc.h`.
+* Add `grpc_channel_create_from_fd` and `grpc_server_add_channel_from_fd` to Core Surface API.
+
+## Background
+
+Currently, the gRPC insecure build does not include any of the functions in `grpc_security.h`, nor does it include any of the code in `src/core/lib/security`. This means that it does not support channel or call credentials, security connectors, or client/server auth filters. It also does not link in SSL libraries.
+
+We could stop supporting insecure builds, so that all builds of gRPC include the functions in `grpc_security.h` and the code in `src/core/lib/security`.  However, the gRPC build would not directly include the SSL libraries.  Instead, there would be separate build targets for each of the credential types (both insecure and secure). Only those credential types that need SSL would pull in the SSL library dependencies.
```

Comment:
Please change the wording here to make it clear that we are actually making these changes.  Instead of "could" or "would", please use "will".

We should also explain why we're doing this.  Key points:
- The current approach requires different API entry points for secure vs. insecure channels and servers, and it requires a lot of conditional compilation that complicates our build system.
- The main reason for supporting insecure builds is because there are some users that don't need security but really need to minimize binary size (e.g., embedded systems), so they need a way to build without depending on the SSL libraries.
- The current approach was decided upon early in our development, before we had the credential APIs, and it was never updated when the credential APIs were added.  However, the credential APIs provide a much cleaner way to allow building gRPC without the SSL dependency.

To solve this problem, we are going to move to an approach where insecure channels and servers are handled the same way as secure channels and servers.  It will no longer be possible to build gRPC without support for credentials or to create a channel or server without specifying credentials, but we will provide an insecure credentials type to be used in cases where security is not needed.  The insecure credentials type will not depend on SSL libraries and will be included in all gRPC builds.  Other credential types that do depend on SSL libraries will be available in secure builds only, just as they are today.

Reply:
Thanks for adding more contexts. I added them to the new text.